lib/TuyaOAuth2Client.js

```diff
@@ -289,22 +300,38 @@ class TuyaOAuth2Client extends OAuth2Client {
               const key = message.headers['x-tuya-key'];
 
               const registeredDevice = this.registeredDevices.get(key);
-              if (!registeredDevice) return;
+              const registeredOtherDevice = this.registeredOtherDevices.get(key);
+              if (!registeredDevice && !registeredOtherDevice) return;
 
               Promise.resolve().then(async () => {
                 switch (message.body.event) {
                   case 'status': {
                     if (!Array.isArray(message.body.data.deviceStatus)) return;
 
-                    await registeredDevice.onStatus(message.body.data.deviceStatus);
+                    if (registeredDevice) {
+                      await registeredDevice.onStatus(message.body.data.deviceStatus);
+                    }
```

Comment:
```suggestion
                      await registeredDevice?.onStatus(message.body.data.deviceStatus);
```

Not sure if this works well together with await, but it looks a lot nicer (or it this Typescript only perhaps?).

Reply:
As discussed in person, the optional chaining is easier to miss than the if-statement, so they will remain unchanged.